text/3697-declarative-attribute-macros.md

```diff
@@ -0,0 +1,230 @@
+- Feature Name: `declarative_attribute_macros`
+- Start Date: 2024-09-20
+- RFC PR: [rust-lang/rfcs#3697](https://github.com/rust-lang/rfcs/pull/3697)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Support defining `macro_rules!` macros that work as attribute macros.
+
+# Motivation
```

Comment:
Thought of another case where this would be very nice; docs reuse. If a paragraph or section is repeated in many locations, it should be possible to put this into an attribute `macro_rules!` and add `#[some_docs]` wherever it is needed.

Thinking about in `std` how floats have nearly identical docs copied+pasted to all four types, compared to integers where a single macro is used that makes finding the actual source for implementations difficult. Putting docs in attribute macros seems like a happy middle ground.